### PR TITLE
[BRE-827] Enable reference status update to handle autofill

### LINKED ISSUE #2851: Autocompletion on Staging/Deploy page leaves box red / keeps error message
Howdy,

I've noticed that when using the autocomplete in the staging deployment tab the error message of there not being a matching branch stays. If you type out the entire branch name, this goes away. 
### Autocompleted
screen shot 2018-07-13 at 4 19 56 pm

### Typed Out
screen shot 2018-07-13 at 4 22 30 pm

### PR DESCRIPTION
**Pull request by ragurney**
Fixes #2851 

/cc @zendesk/samson

 Jira link: [BRE-827](https://zendesk.atlassian.net/browse/BRE-827)

### CODE REVIEW

File: app/assets/javascripts/ref_status_typeahead.js [hunk 1]
```diff
@@ -81,7 +81,7 @@ function refStatusTypeahead(options){
   initializeTypeahead();
 
   // TODO: clean up by wrapping this in a limiter function
-  $reference.on('input', function() {
+  $reference.onAllValueChanges(function() {
```
Comment (grosser) [inline app/assets/javascripts/ref_status_typeahead.js]: should be a more obvious name like `pollForChange(100, function)`

File: app/assets/javascripts/on_all_changes.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+  var pollForChange = function () {
+    var newValue = $(self).val();
+    if (newValue !== lastValue) {
+      lastValue = newValue;
+      callback.call(self);
+    }
+    setTimeout(pollForChange, 100);
+  };
+  setTimeout(pollForChange, 100);
```
Comment (grosser) [inline app/assets/javascripts/on_all_changes.js]: setInterval ?

File: app/assets/javascripts/ref_status_typeahead.js [hunk 3]
```diff
@@ -81,7 +81,7 @@ function refStatusTypeahead(options){
   initializeTypeahead();
 
   // TODO: clean up by wrapping this in a limiter function
```
Comment (grosser) [inline app/assets/javascripts/ref_status_typeahead.js]: can remove the TODO then ?

**Comment by grosser** [comment]
add a comment explaining why this is done/which cases need to be considered when refactoring

File: app/assets/javascripts/on_all_changes.js (hunk 2 above)
Comment (ragurney) [reply app/assets/javascripts/on_all_changes.js]: ✨

**Comment by ragurney** [comment]
Good to go?

File: app/assets/javascripts/ref_status_typeahead.js [hunk 4]
```diff
@@ -80,8 +80,8 @@ function refStatusTypeahead(options){
 
   initializeTypeahead();
 
-  // TODO: clean up by wrapping this in a limiter function
-  $reference.on('input', function() {
+  // Continuously polling for change to account for autofill which does not trigger input/change events
```
Comment (grosser) [inline app/assets/javascripts/ref_status_typeahead.js]: ✨

File: app/assets/javascripts/on_all_changes.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+$.fn.pollForChange = function (interval, callback) {
```
Comment (grosser) [inline app/assets/javascripts/on_all_changes.js]: method name does not match file name

File: app/assets/javascripts/on_all_changes.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+  var self = this;
+  var lastValue = "";
+
+  var pollForChange = function () {
```
Comment (grosser) [inline app/assets/javascripts/on_all_changes.js]: could inline this as `setInterval(function(){ .... })`

**Pull request merged**